csrc/kernel_ir.cpp

```diff
@@ -627,7 +627,8 @@ bool ForLoop::isTrivial() const {
   }
 
   // Extent-1 loop: for (int i = 0; i < 1; ++i) {
-  if (start()->isZeroInt() && stop()->isOneInt() && step()->isOneInt()) {
+  if (start()->isZeroInt() && simplifiedStop()->isOneInt() &&
+      step()->isOneInt()) {
```

Comment:
Ideally, this should be `simplifyExpr(start < end && start + step >= end)->getBool() == true`, but I don't see any such case, so I am just keeping this narrow case.